### PR TITLE
📝 Reset cookiecutter + Licence notice

### PR DESCRIPTION
**Pull request by acabarbaye**
### Description

Added licence notice and reset with cookie cutter

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [x]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,7 +24,7 @@
 
   - id: mypy
     name: mypy
-    entry: mypy -p {{ cookiecutter.python_module_name }}
```
Comment (muchzill4) [inline .pre-commit-config.yaml]: WELP

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,12 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
 [ProjectConfig]
 # Only path variables can and must contain 'DIR', 'PATH' or 'ROOT' in their name as
 # these tokens are used to identify path variables from other variable types.
 PROJECT_ROOT = "."
+PROJECT_NAME = "Mbed devices"
```
Comment (muchzill4) [inline pyproject.toml]: Why is this required? Would I be a terrible nitpicker if I asked for it to be consistent with `README.md` where we call this `Mbed Devices` (both uppercase)?

Comment (acabarbaye) [reply pyproject.toml]: My mistake.
```suggestion
PROJECT_NAME = "Mbed Devices"
```
